src/Utilities/Compiler/CodeMetrics/MetricsHelper.cs

```diff
@@ -86,11 +86,42 @@ internal static async Task<long> GetLinesOfCodeAsync(ImmutableArray<SyntaxRefere
                     // Declaration on a single line, we count it as a separate line.
                     delta = 1;
                 }
+                else
+                {
+                    // Ensure that we do not count the leading and trailing emtpy new lines.
+                    var additionalNewLines = Math.Max(0, GetNewlineCount(declSyntax.GetLeadingTrivia(), leading: true) + GetNewlineCount(declSyntax.GetTrailingTrivia(), leading: false) - 1);
+                    delta -= additionalNewLines;
+                }
 
                 linesOfCode += delta;
             }
 
             return linesOfCode;
+
+            static int GetNewlineCount(SyntaxTriviaList trivialList, bool leading)
+            {
+                var triviaParts = trivialList.ToFullString().Split(new[] { "\r\n", "\r", "\n" }, StringSplitOptions.None).ToImmutableArray();
+                return GetNewlineCount(triviaParts, leading);
+
+                static int GetNewlineCount(ImmutableArray<string> triviaParts, bool leading)
+                {
+                    var index = leading ? 0 : triviaParts.Length - 1;
+                    var loopCondition = leading ? LoopConditionForLeading : (Func<int, int, bool>)LoopConditionForTrailing;
+                    var incrementOrDecrement = leading ? 1 : -1;
+                    var count = 0;
+                    for (; loopCondition(index, triviaParts.Length); index += incrementOrDecrement, count++)
+                    {
+                        var trivia = triviaParts[index].Trim();
+                        if (trivia.Length != 0)
+                            break;
```

Comment:
A couple of questions:
1. Looks like we count comments as part of "Lines"?
2. Do we want to exclude empty newlines in between comments too?

Reply:
I thought about it, but I think it is reasonable to count the start of any non-whitespace trivia as a line belonging to next declaration.

Reply:
Whatev the answer is, should have a test with blank lines between comments.

Reply:
Sure, added test. We can go both ways here, but this seems to be a rare case, so we can revisit if we get feedback this is problematic.